### PR TITLE
 [MTDSA-1072] Update APIs #43 and #45 with the latest DES specs

### PR DESCRIPTION
**Pull request by dabd**

### CODE REVIEW

File: app/uk/gov/hmrc/selfassessmentapi/connectors/PropertiesPeriodConnector.scala [hunk 1]
```diff
@@ -81,6 +81,6 @@ object PropertiesPeriodConnector {
   def retrieveAll(nino: Nino, propertyType: PropertyType)(
       implicit hc: HeaderCarrier): Future[PropertiesPeriodResponse] =
     httpGet[PropertiesPeriodResponse](
-      baseUrl + s"/income-store/nino/$nino/uk-properties/$propertyType/periodicsummaries",
+      baseUrl + s"/income-store/nino/$nino/uk-properties/$propertyType/periodic-summaries",
```
Comment (GlassAndOneHalf) [inline app/uk/gov/hmrc/selfassessmentapi/connectors/PropertiesPeriodConnector.scala]: Is this released now? The reason the url is `periodicsummaries` instead of `periodic-summaries` is because there is a route conflict in the DES simulator.

File: app/uk/gov/hmrc/selfassessmentapi/resources/PropertiesAnnualSummaryResource.scala [hunk 2]
```diff
@@ -64,9 +64,7 @@ object PropertiesAnnualSummaryResource extends BaseResource {
                 }
               case None => NotFound
             }
-          case 404 => NotFound
-          case 400 => BadRequest(Error.from(response.json))
-          case _ => unhandledResponse(response.status, logger)
```
Comment (GlassAndOneHalf) [inline app/uk/gov/hmrc/selfassessmentapi/resources/PropertiesAnnualSummaryResource.scala]: Why did you remove the unhandled response case? If you want to remove this you need to add logging to `Error.from2`. Also, if the unhandled response doesn't contain JSON, this will throw an exception.

File: app/uk/gov/hmrc/selfassessmentapi/resources/PropertiesPeriodResource.scala [hunk 3]
```diff
@@ -111,9 +112,7 @@ object PropertiesPeriodResource extends BaseResource {
               case PropertyType.OTHER =>
                 ResponseMapper[Other.Properties, des.properties.Other.Properties].allPeriods(response, getMaxPeriodTimeSpan)
             }))
-          case 400 => BadRequest(Error.from(response.json))
-          case 404 => NotFound
-          case _ => unhandledResponse(response.status, logger)
```
Comment (GlassAndOneHalf) [inline app/uk/gov/hmrc/selfassessmentapi/resources/PropertiesPeriodResource.scala]: Same as above.

File: func/uk/gov/hmrc/support/BaseFunctionalSpec.scala [hunk 4]
```diff
@@ -1270,7 +1270,7 @@ trait BaseFunctionalSpec extends TestApplication {
 
         def periodsWillBeReturnedFor(nino: Nino, propertyType: PropertyType): Givens = {
           stubFor(
-            get(urlEqualTo(s"/income-store/nino/$nino/uk-properties/$propertyType/periodicsummaries"))
+            get(urlEqualTo(s"/income-store/nino/$nino/uk-properties/$propertyType/periodic-summaries"))
```
Comment (GlassAndOneHalf) [inline func/uk/gov/hmrc/support/BaseFunctionalSpec.scala]: Same as initial comment, if the des simulator route conflict is not fixed, this change can not be applied.

File: app/uk/gov/hmrc/selfassessmentapi/connectors/PropertiesPeriodConnector.scala (hunk 1 above)
Comment (rakeshavasarala) [inline app/uk/gov/hmrc/selfassessmentapi/connectors/PropertiesPeriodConnector.scala]: but the actual DES url is /income-store/nino/$nino/uk-properties/$propertyType/periodic-summaries
